### PR TITLE
V2.0 `tbl_ard_wide_summary()`

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
Added function `tbl_ard_wide_summary()`

**If there is an GitHub issue associated with this pull request, please provide link.**

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed by running `renv::install()`
- [ ] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: R/tbl_ard_wide_summary.R [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    append(
+      list(
+        cards = list(tbl_ard_wide_summary = cards),
+        inputs = tbl_ard_wide_summary_inputs,
+        call_list <- list(tbl_ard_wide_summary = call)
+      )
+    ) |>
+    structure(class = c("tbl_ard_wide_summary", "gtsummary"))
+
+  # return tbl_summary table ---------------------------------------------------
```
Comment (Melkiades) [inline R/tbl_ard_wide_summary.R:141]: Should there be some theme handling here?

File: R/tbl_wide_summary.R [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    # fill in any types not specified by user
+    type <- utils::modifyList(default_types, type)
+  } else {
+    type <- assign_summary_type(data, include, value)
+  }
+
+  # summary types must be all 'continuous' OR c('categorical', 'dichotomous')
+  if (!all(unlist(type) %in% "continuous") && !all(unlist(type) %in% c("categorical", "dichotomous"))) {
+    cli::cli_abort(
+      "The summary types must all be {.val continuous} or all be {.val {c('categorical', 'dichotomous')}}.",
```
Comment (Melkiades) [inline R/tbl_wide_summary.R:69]: I think either here we suggest doing something of the following sort or we allow to do the following sort:
```r
a <- tbl_wide_summary(
  trial,
  include = c(response, death, grade, stage)
)

b <- tbl_wide_summary(
  trial,
  include = c(age, marker)
)

tbl_merge(list(a, b), tab_spanner = FALSE) |>
  print()

```

File: R/tbl_ard_wide_summary.R [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+#'   ard_categorical(variables = grade),
+#'   .missing = TRUE,
+#'   .attributes = TRUE
+#' ) |>
+#'   tbl_ard_wide_summary()
+tbl_ard_wide_summary <- function(cards,
+                                 statistic =
+                                   switch(type[[1]],
+                                          "continuous" = c("{median}", "{p25}, {p75}"),
+                                          c("{n}", "{p}%")),
```
Comment (Melkiades) [inline R/tbl_ard_wide_summary.R:39]: Just playing with this, I got here:
```r
trial1 <- head(trial)
bind_ard(
  ard_continuous(trial1, variables = c(age, marker)),
  ard_categorical(trial1, variables = c(age, marker)),
  ard_attributes(trial1, variables = c(age, marker))
) |>
  brdg_wide_summary(
    variables = c("age", "marker"),
    statistic = list(age = c("{mean}", "{sd}", "{p}%"), marker = c("{mean}", "{sd}", "{p}%")),
    type = list(age = "continuous", marker = "continuous")
  )
```
Here only the first is called in. I was wondering if we should throw in an error or allow it with a warning or not allow it. Then would it make sense to NOT enforce the either categorical/dichotomous either numeric?

**Review by Melkiades — APPROVED** [review]
I think it is very good! My comments are mainly about how much we need to align with `tbl_summary` inputs and so on. I am also wondering we should also integrate something similar for custom functions (like `tbl_custom_summary`)? But we can discuss this afterwards if you think it is inline with our plans!!

File: R/tbl_ard_wide_summary.R [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                                 type = NULL,
+                                 value = NULL,
+                                 include = everything()) {
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/tbl_ard_wide_summary.R:42]: The sorting happens only when we have a data frame (we make the variable a factor and order the levels based on frequency). We could certainly add something in the future for this. But I think it's ok for now.

I wasn't sure how the missing should look in a table like this, so I just left it out :)

File: R/tbl_wide_summary.R (hunk 2 above)
Comment (ddsjoberg) [reply R/tbl_wide_summary.R:69]: I am not 100% sure i follow the comment...but let's discuss next week!

File: R/tbl_ard_wide_summary.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (Melkiades) [reply R/tbl_ard_wide_summary.R:42]: it makes sense! we can add it at a later stage when it is a needed feature

File: R/tbl_ard_wide_summary.R (hunk 1 above)
Comment (Melkiades) [reply R/tbl_ard_wide_summary.R:141]: I agree 100%. The code is quite modular, and this is easy to add later!

**Pull request merged**